### PR TITLE
Seperate compute_returns_and_advantage

### PR DESCRIPTION
**Pull request by hades-rp2010**
Wrt #320 

Preferably merge after #307 (Merging 307 will remove the conflicts in this PR)

### CODE REVIEW

**Comment by hades-rp2010** [comment]
> Wrt #320
> 
> Preferably merge after #307 (Merging 307 will remove the conflicts in this PR)

My bad, merge conflicts are resolved

File: genrl/core/rollout_storage.py [hunk 1]
```diff
@@ -310,3 +257,66 @@ def _get_samples(self, batch_inds: np.ndarray) -> RolloutBufferSamples:
             self.returns[batch_inds].flatten(),
         )
         return RolloutBufferSamples(*tuple(map(self.to_torch, data)))
+
+
+def compute_returns_and_advantage(
+    rollout_buffer: Union[RolloutBuffer, BaseBuffer],
+    last_value: torch.Tensor,
+    dones: np.ndarray,
+    use_gae: bool = False,
```
Comment (Sharad24) [inline genrl/core/rollout_storage.py:266]: Put a little more thought into this. Its trivial to separate the function out directly.

**Review by Sharad24 — COMMENTED** [review]
Should this function even be part of rollout_storage now?

**Comment by hades-rp2010** [comment]
> Should this function even be part of rollout_storage now?

I think it makes more sense to shift this function inside the OnPolicyAgent now

**Comment by hades-rp2010** [comment]
@Sharad24 If no other changes to this are reqd, this could be merged as well

**Comment by Sharad24** [comment]
Can you sync this with master? Will merge it after that.

**Pull request merged**